spec/saml_response_spec.rb

```diff
@@ -0,0 +1,104 @@
+require_relative "spec_helper"
+require_relative "../lib/fake_idp/saml_response"
+require "ruby-saml"
+require "securerandom"
+
+RSpec.describe FakeIdp::SamlResponse do
+  before do
+    FakeIdp.configure do |config|
+      config.callback_url = "http://localhost.dev:3000/auth/saml/devidp/callback"
+      config.sso_uid = "12345"
+      config.username = "bobthessouser"
+      config.name_id = "bobthessouser@example.com"
+      config.first_name = "Reid"
+      config.last_name = "Smith"
+      configuration.email = "reid@msn.com"
+      config.idp_certificate = fake_public_certificate
+      config.idp_secret_key = fake_private_key
+      config.algorithm = :sha1
+    end
+  end
+
+  let(:configuration) { FakeIdp.configuration }
+
+  it "generates a valid SAML response" do
+    settings = OneLogin::RubySaml::Settings.new(
+      name_identifier_format: "urn:oasis:names:tc:SAML:1.1:nameid-format:unidentified",
+      allowed_clock_drift: 10000000,
```

Comment:
What's the significance here?

Reply:
@cbortz The ruby-saml gem does a time based validation for the `NotOnOrAfter` attributes in the SAML response. Responses usually have these set to a few minutes after when the request originated, setting how long that request is valid. If the gem sees a response with those attributes outside an acceptable time frame, the response becomes invalid and is rejected. The `allowed_clock_drift` option tells the gem to ignore that validation for the passed number of seconds and is done here since we're not concerned with validating that for this test.